server/channels/store/searchlayer/layer.go

```diff
@@ -111,6 +111,32 @@ func (s *SearchStore) indexUser(rctx request.CTX, user *model.User) {
 	}
 }
 
+func (s *SearchStore) indexChannelsForTeam(rctx request.CTX, teamID string) {
```

Comment:
Do we not index DMs and GMs or are they handled elsewhere?

Reply:
DM/GMs are not tied to a team, so they don't need any change. Same for private channels. A user cannot access any private channels immediately after joining a team unless they are added to it.